src/dodal/devices/pressure_jump_cell_adc.py

```diff
@@ -0,0 +1,63 @@
+from enum import Enum
+from functools import partial
+
+from ophyd_async.core import (
+    ConfigSignal,
+    DeviceVector,
+    SignalR,
+    StandardReadable,
+)
+from ophyd_async.epics.signal import epics_signal_r, epics_signal_rw
+
+from ophyd import ADComponent as ADC
+from ophyd import (
+    AreaDetector,
+    CamBase,
+    Component,
+    Device,
+    EpicsSignal,
+    HDF5Plugin,
+    OverlayPlugin,
+    ProcessPlugin,
+    ROIPlugin,
+    StatsPlugin,
+    Signal,
+    StatusBase,
+)
+
+class PressureJumpAdcConfigParams:
+    pass
+
+class PressureJumpCellADC(AreaDetector):
+    """
+    High pressure X-ray cell fast ADC, used to capture pressure jumps.
+    """
+    cam = ADC(CamBase, "-EA-HPXC-01:CAM:")
+    scale = ADC(ROIPlugin, "-EA-HPXC-01:SCALE:")
+    # trig = # TODO support module adUtil NDPluginReframe? 
+    # arr = # TODO support module ADCore NDPluginStdArrays? required?
+    hdf5 = ADC(HDF5Plugin, "-DI-OAV-01:HDF5:")
+
+    p1Roi = ADC(ROIPlugin, "-EA-HPXC-01:ROIP1:")
+    p1Proc = ADC(ProcessPlugin, "-EA-HPXC-01:PROCP1:")
+    p1Stats = ADC(StatsPlugin, "-EA-HPXC-01:STATP1:")
+
+    p2Roi = ADC(ROIPlugin, "-EA-HPXC-01:ROIP2:")
+    p2Proc = ADC(ProcessPlugin, "-EA-HPXC-01:PROCP2:")
+    p2Stats = ADC(StatsPlugin, "-EA-HPXC-01:STATP2:")
+
+    p3Roi = ADC(ROIPlugin, "-EA-HPXC-01:ROIP3:")
+    p3Proc = ADC(ProcessPlugin, "-EA-HPXC-01:PROCP3:")
+    p3Stats = ADC(StatsPlugin, "-EA-HPXC-01:STATP3:")
+
+    def __init__(self, *args, params: PressureJumpAdcConfigParams, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.parameters = params
+        self.snapshot.oav_params = params
+        self.subscription_id = None
+        self._snapshot_trigger_subscription_id = None
+
+    def wait_for_connection(self, all_signals=False, timeout=2):
+        connected = super().wait_for_connection(all_signals, timeout)
+        return connected
+
```

Comment:
Compare the implementation of e.g. the AdAravis in ophyd-async
https://github.com/bluesky/ophyd-async/tree/e4fb8492bfad29dff9f539f6adc63c8152f68493/src/ophyd_async/epics/adaravis

This should be EthercatADC(StandardDetector) with an EthercatController, EthercatDriver if appropriate (from the looks of this can just use AdBase though, as no special signals?)

Reply:
Thank you Joseph I'll have a look.  At the moment I'm not sure what would make a signal 'special'.

Reply:
Are there any signals for the pressure exposed on the camera itself? From the ophyd definition I don't see any signals that are unique to function of the device class or particularly sought out for the experimental procedure, which is what I mean by "special".

Reply:
Ah ok, looking at the ioc configuration for the pressure jump cell BL38P-EA-IOC-12  it uses ADEthercat, ADCore, and adUtil standard support modules where there are 4 ethercat channels likely for the pressures. I wasn't sure if this will be picked up from the IOC by the ophyd modules or whether this needs to be specified in the ophyd device? 

There is an additional support module specific to the pressure cell where `pressureCell.scale` is used in the IOC configuration which I presume adjusts the scale for the visualisation.

Reply:
Updated to use ophyd-async in 5f9f808cbd5260e22fc69b09b8524aabc9ea0434.